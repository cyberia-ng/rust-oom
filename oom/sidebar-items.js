initSidebarItems({"struct":[["NonEmptyMutSlice","A non-empty mutable slice type, counterpart of `&mut [T]`."],["NonEmptySlice","A non-empty slice type, counterpart of `&[T]`."],["NonEmptyVec","A non-empty vector type, counterpart of `Vec<T>`."]]});