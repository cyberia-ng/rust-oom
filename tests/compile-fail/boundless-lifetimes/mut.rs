use oom::NonEmptyMutSlice;
fn main() {
    let mut foo = String::from("hello");
    let mut s = NonEmptyMutSlice::from_mut(&mut foo);
    let bad: &'static [_] = s.as_mut_slice();
    drop(foo);
    let _dangling = bad.first();
}
