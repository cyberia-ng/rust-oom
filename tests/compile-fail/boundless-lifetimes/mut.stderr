error[E0597]: `foo` does not live long enough
 --> $DIR/mut.rs:4:44
  |
4 |     let mut s = NonEmptyMutSlice::from_mut(&mut foo);
  |                                            ^^^^^^^^ borrowed value does not live long enough
5 |     let bad: &'static [_] = s.as_mut_slice();
  |              ------------ type annotation requires that `foo` is borrowed for `'static`
...
8 | }
  | - `foo` dropped here while still borrowed

error[E0597]: `s` does not live long enough
 --> $DIR/mut.rs:5:29
  |
5 |     let bad: &'static [_] = s.as_mut_slice();
  |              ------------   ^ borrowed value does not live long enough
  |              |
  |              type annotation requires that `s` is borrowed for `'static`
...
8 | }
  | - `s` dropped here while still borrowed

error[E0505]: cannot move out of `foo` because it is borrowed
 --> $DIR/mut.rs:6:10
  |
4 |     let mut s = NonEmptyMutSlice::from_mut(&mut foo);
  |                                            -------- borrow of `foo` occurs here
5 |     let bad: &'static [_] = s.as_mut_slice();
  |              ------------ type annotation requires that `foo` is borrowed for `'static`
6 |     drop(foo);
  |          ^^^ move out of `foo` occurs here
